src/Sparrow.Server/ByteString.cs

```diff
@@ -841,21 +841,108 @@ public void Reset()
 
                 ReleaseSegment(segment);
             }
-            
+
             _wholeSegments.Clear();
             _wholeSegments.Add(_internalCurrent);
             _wholeSegments.Add(_externalCurrent);
         }
 
+        public void DefragmentSegments()
+        {
+            // small allocators
+            if (_totalAllocated < 128 * Sparrow.Global.Constants.Size.Megabyte)
+                return;
+
+            // small fragmentation
+            var segments = _internalReadyToUseMemorySegments;
+            if (segments == null || segments.Count < 1024)
+                return;
+
+            var newSegments = MergeSegments(segments, ascending: true);
+            newSegments = MergeSegments(newSegments, ascending: false);
```

Comment:
Why do we need to search forward and backward?

Reply:
Answered here: https://github.com/ravendb/ravendb/pull/15590#discussion_r1066705336